presto-client/src/main/java/com/facebook/presto/client/ClientSession.java

```diff
@@ -88,6 +92,27 @@ public static ClientSession withProperties(ClientSession session, Map<String, St
                 session.getTimeZoneId(),
                 session.getLocale(),
                 properties,
+                session.getPreparedStatements(),
+                session.getTransactionId(),
+                session.isDebug(),
+                session.getClientRequestTimeout());
+    }
+
+    public static ClientSession withPreparedStatements(ClientSession session, Map<String, String> preparedStatements)
```

Comment:
huh, it was working because sessionStatements was never used.  which is it's own issue.  I've fixed it now.